index.js

```diff
@@ -264,23 +296,47 @@ module.exports = class Combobox {
   }
 
   select() {
+    let newSelected = false;
     const currentOpt = this.currentOption;
     if (!currentOpt) { return; }
-    if (!this.multiselect && this.selected) { // clean up previously selected
-      this.selected.classList.remove('selected');
+
+    if (!this.config.multiselect && this.selected.length) { // clean up previously selected
+      Classlist(this.selected[0]).remove(this.config.selectedClass)
+    }
+
+    // Multiselect option
+    if (this.config.multiselect) {
+      const idx = this.selected.indexOf(currentOpt);
+      //If option is in array and gets clicked, remove it
+      if (idx > -1) {
+        this.selected.splice(idx, 1);
+      } else {
+        this.selected.push(currentOpt);
+      }
+    } else {
+      // Single select stuff
+      this.selected = [currentOpt];
     }
 
-    currentOpt.classList.add(this.config.selectedClass);
-    this.selected = currentOpt;
+    // Taking care of adding / removing selected class
+    if (currentOpt.classList.contains(this.config.selectedClass)) {
```

Comment:
for older browser support and consistency, use `Classlist(currentOpt).contains(this.config.selectedClass)`